src/js/utils/events.js

```diff
@@ -0,0 +1,47 @@
+// using a symbol to avoid "discoverable" private props
+const listeners = Symbol('listeners');
+
+export class Observable {
+    constructor() {
+        this[listeners] = new Map();
+    }
+
+    on(event, handler) {
+        let map = this[listeners];
+        if (!map.has(event)) {
+            map.set(event, new Set());
+        }
+
+        map.get(event).add(handler);
+    }
+
+    off(event, handler) {
+        let map = this[listeners];
+        if (map.has(event)) {
+            map.get(event).delete(handler);
+        }
+    }
+
+    trigger(event, args) {
```

Comment:
maybe it shouldn't be called args.. I thought about doing that, but the current eventing system (backbone events) use `(event, options)`